admin-dev/themes/new-theme/js/pages/product/edit/virtual-product-manager.ts

```diff
@@ -76,4 +77,33 @@ export default class VirtualProductManager {
   private showContent(): void {
     this.$fileContentContainer.removeClass('d-none');
   }
+
+  private fillVirtualProductNameField(): void {
+    $(ProductMap.virtualProduct.fileUploadField).on('change', () => {
+      // do not fill the name field if it's not empty
+      if ($(ProductMap.virtualProduct.fileNameField).val() !== '') {
+        return;
+      }
+
+      let fullPath = $(ProductMap.virtualProduct.fileUploadField).val();
+
+      if (Array.isArray(fullPath) || typeof fullPath === 'undefined') {
+        throw new Error('Full path of the file is not supposed to be an array');
```

Comment:
If you want to throw there, you should try catch the execution, overwise it will break the JS execution for the whole page and it's probably something we don't want

Reply:
Why not a `console.error` instead then?